discovery-infra/test_infra/assisted_service_api.py

```diff
@@ -145,11 +146,21 @@ def get_hosts_in_error_status(self, cluster_id: str):
     def clusters_list(self) -> List[Dict[str, Any]]:
         return self.client.list_clusters()
 
+    def infra_envs_list(self) -> List[models.InfraEnv]:
+        return self.client.list_infra_envs()
+
     def get_all_clusters(self) -> List[Dict[str, Any]]:
         return self.client.list_clusters(get_unregistered_clusters=True)
 
     def cluster_get(self, cluster_id: str) -> models.cluster.Cluster:
         return self.client.get_cluster(cluster_id=cluster_id)
+    
+    def get_infra_env_by_cluster_id(self, cluster_id: str) -> InfraEnv:
```

Comment:
if we don't find any, do we want to return ``None`` or to raise a ``ValueError`` exception?